sdk/python/feast/pyspark/launchers/standalone/local.py

```diff
@@ -215,42 +219,56 @@ def historical_feature_retrieval(
         self, job_params: RetrievalJobParameters
     ) -> RetrievalJob:
         job_id = str(uuid.uuid4())
-        return StandaloneClusterRetrievalJob(
+        job = StandaloneClusterRetrievalJob(
             job_id,
             job_params.get_name(),
             self.spark_submit(job_params),
             job_params.get_destination_path(),
         )
+        JOB_CACHE[job_id] = job  # type: ignore
```

Comment:
Do we really have to ignore types?

Reply:
Yes, mypy is acting up because it's detecting the JOB_CACHE by first assignment of job and it's complaining on the second assignment (they have different types, e.g. RetrievalJob vs BatchIngestionJob). mypy isn't able to detect the shared parent class unfortunately.

Reply:
Does Union[TypeA, TypeB] work?

Reply:
I thought it wasn't possible to define types of global variables, seems like I was incorrect. Defining the type explicitly helped. Will change this too.

Reply:
What I meant about inheritance is that it's not able to "update" the type dynamically. e.g. if mypy infers from 1 assignment that the type is Dict[str, B], and from 2nd assignment it sees that type is Dict[str, C], while B & C are subclasses of A, it's not able to update the type to Dict[str, A].